flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/FlinkSchemaUtil.java

```diff
@@ -64,26 +68,75 @@ public static Schema convert(TableSchema schema) {
+  /** Convert the flink table schema to apache iceberg schema with column comment. */
+  public static Schema convert(ResolvedSchema flinkSchema) {
+    List<Column> tableColumns = flinkSchema.getColumns();
+    // copy from org.apache.flink.table.api.Schema#toRowDataType
+    DataTypes.Field[] fields =
+        tableColumns.stream()
+            .map(
+                column -> {
+                  if (column.getComment().isPresent()) {
+                    return DataTypes.FIELD(
+                        column.getName(), column.getDataType(), column.getComment().get());
+                  } else {
+                    return DataTypes.FIELD(column.getName(), column.getDataType());
+                  }
+                })
+            .toArray(DataTypes.Field[]::new);
+
+    LogicalType schemaType = DataTypes.ROW(fields).notNull().getLogicalType();
+    Preconditions.checkArgument(
+        schemaType instanceof RowType, "Schema logical type should be RowType.");
+
+    RowType root = (RowType) schemaType;
+    Type converted = root.accept(new FlinkTypeToType(root));
+    Schema icebergSchema = new Schema(converted.asStructType().fields());
+    return freshIdentifierFieldIds(icebergSchema, flinkSchema);
+  }
+
+  /** @deprecated Use {@link #freshIdentifierFieldIds(Schema, ResolvedSchema)} instead. */
+  @Deprecated
+  private static Schema freshIdentifierFieldIds(Schema icebergSchema, TableSchema flinkSchema) {
+    // Locate the identifier field id list.
+    Set<Integer> identifierFieldIds = Sets.newHashSet();
+    if (flinkSchema.getPrimaryKey().isPresent()) {
+      for (String column : flinkSchema.getPrimaryKey().get().getColumns()) {
+        Types.NestedField field = icebergSchema.findField(column);
+        Preconditions.checkNotNull(
+            field,
+            "Cannot find field ID for the primary key column %s in flinkSchema %s",
+            column,
+            icebergSchema);
+        identifierFieldIds.add(field.fieldId());
+      }
+    }
+
+    return new Schema(
+        icebergSchema.schemaId(), icebergSchema.asStruct().fields(), identifierFieldIds);
+  }
+
+  private static Schema freshIdentifierFieldIds(Schema icebergSchema, ResolvedSchema flinkSchema) {
     // Locate the identifier field id list.
     Set<Integer> identifierFieldIds = Sets.newHashSet();
-    if (schema.getPrimaryKey().isPresent()) {
-      for (String column : schema.getPrimaryKey().get().getColumns()) {
-        Types.NestedField field = iSchema.findField(column);
+    if (flinkSchema.getPrimaryKey().isPresent()) {
+      for (String column : flinkSchema.getPrimaryKey().get().getColumns()) {
+        Types.NestedField field = icebergSchema.findField(column);
         Preconditions.checkNotNull(
             field,
-            "Cannot find field ID for the primary key column %s in schema %s",
+            "Cannot find field ID for the primary key column %s in flinkSchema %s",
```

Comment:
this seems incorrect. I think the old comment is correct, `schema` implied `Iceberg table schema`.

also style is to avoid variable name in error message, which is not user friendly. hence it would have been `Flink schema`.